From 364f65bc0bd61e314e7d813d74648e8799f0f7e5 Mon Sep 17 00:00:00 2001
From: Victor Wan <victor.wan@amlogic.com>
Date: Wed, 6 Aug 2014 10:28:33 +0800
Subject: [PATCH 4673/5965] cpufreq, camera: fix section mismatch warning

remove below 3 warnings:

WARNING: vmlinux.o(.text+0x355f94): Section mismatch in reference from the function handle_up_cpu() to the function .cpuinit.text:cpu_up()
The function handle_up_cpu() references
the function __cpuinit cpu_up().
This is often because handle_up_cpu lacks a __cpuinit
annotation or the annotation of cpu_up is wrong.

WARNING: vmlinux.o(.data+0x608c4): Section mismatch in reference from the variable cam_devs to the function .init.text:sp1628_v4l2_probe()
The variable cam_devs references
the function __init sp1628_v4l2_probe()
If the reference is valid then annotate the
variable with __init* or __refdata (see linux/init.h) or name the variable:
*_template, *_timer, *_sht, *_ops, *_probe, *_probe_one, *_console

WARNING: vmlinux.o(.data+0x608d8): Section mismatch in reference from the variable cam_devs to the function .init.text:bf3720_v4l2_probe()
The variable cam_devs references
the function __init bf3720_v4l2_probe()
If the reference is valid then annotate the
variable with __init* or __refdata (see linux/init.h) or name the variable:
*_template, *_timer, *_sht, *_ops, *_probe, *_probe_one, *_console
---
 drivers/amlogic/camera/common/cam_prober.c | 4 ++--
 drivers/cpufreq/cpufreq.c                  | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/camera/common/cam_prober.c b/drivers/amlogic/camera/common/cam_prober.c
index 15961dcdae58..573d491ff227 100755
--- a/drivers/amlogic/camera/common/cam_prober.c
+++ b/drivers/amlogic/camera/common/cam_prober.c
@@ -480,7 +480,7 @@ int ar0833_v4l2_probe(struct i2c_adapter *adapter)
 #endif
 
 #ifdef CONFIG_VIDEO_AMLOGIC_CAPTURE_SP1628
-int __init sp1628_v4l2_probe(struct i2c_adapter *adapter)
+int sp1628_v4l2_probe(struct i2c_adapter *adapter)
 {
     int ret = 0;
 	unsigned char reg[2];   
@@ -493,7 +493,7 @@ int __init sp1628_v4l2_probe(struct i2c_adapter *adapter)
 #endif
 
 #ifdef CONFIG_VIDEO_AMLOGIC_CAPTURE_BF3720
-int __init bf3720_v4l2_probe(struct i2c_adapter *adapter)
+int bf3720_v4l2_probe(struct i2c_adapter *adapter)
 {
     int ret = 0;
 	unsigned char reg[2];   
diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index 55b889e62b63..3cbfc8c5a4e4 100755
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -1144,7 +1144,7 @@ static void handle_update(struct work_struct *work)
 	pr_debug("handle_update for cpu %u called\n", cpu);
 	cpufreq_update_policy(cpu);
 }
-static void handle_up_cpu(struct work_struct *work)
+static void __ref handle_up_cpu(struct work_struct *work)
 {
 	int i;
 
-- 
2.19.0

